mev-boost-rs/src/relay_mux.rs

```diff
@@ -125,7 +126,11 @@ impl BlindedBlockProvider for RelayMux {
     ) -> Result<(), BlindedBlockProviderError> {
         let registrations = &registrations;
         let responses = stream::iter(self.relays.iter().cloned())
-            .map(|relay| async move { relay.register_validators(registrations).await })
+            .map(|relay| async move {
+                (|| relay.register_validators(registrations))
```

Comment:
I'm assuming this breaks if the call is not wrapped in a closure ?

Reply:
yup, that's right. it needs smth invokable to be able to retry